odoo/tools/sql.py

```diff
@@ -136,6 +137,19 @@ def convert_column(cr, tablename, columnname, columntype):
         cr.execute(query.format(tablename, columnname, columntype))
     _schema.debug("Table %r: column %r changed to type %s", tablename, columnname, columntype)
 
+def convert_column_translatable(cr, tablename, columnname, columntype):
+    """ Convert the column from/to a 'jsonb' translated field column. """
+    drop_index(cr, f"{tablename}_{columnname}_index", tablename)
+    query = 'ALTER TABLE "{0}" RENAME COLUMN "{1}" TO __temp_type_cast;'
+    query += '\nALTER TABLE "{0}" ADD COLUMN "{1}" {2};'
+    if columntype == 'jsonb':
+        query += '\nUPDATE "{0}" SET "{1}"=json_build_object(\'en_US\', __temp_type_cast::varchar) WHERE __temp_type_cast IS NOT NULL;'
+    else:
+        query += '\nUPDATE "{0}" SET "{1}"=__temp_type_cast->>\'en_US\'::{2};'
+    query += '\nALTER TABLE "{0}" DROP COLUMN __temp_type_cast CASCADE;'
```

Comment:
This is not optimal. There is an SQL syntax for converting data when changing column type.
```suggestion
    if columntype == "jsonb":
        using = 'CASE WHEN "{columnname}" IS NOT NULL THEN json_build_object(\'en_US\', "{columname}"::varchar) END'
    else:
        using = f'"{columnname}"->>\'en_US\''
    
    query = f'ALTER TABLE "{tablename}" ALTER COLUMN "{columname}" TYPE {columntype} USING {using}"'
```

At least PostgreSQL can optimize the change. Else it's just a plain `UPDATE`.